src/block_cache/block_cache.c

```diff
@@ -498,7 +540,13 @@ __wti_blkcache_put(
 
 err:
     __blkcache_free(session, data_ptr);
-    __blkcache_free(session, blkcache_store);
+    if (blkcache_deltas != NULL) {
```

Comment:
Isn't this check redundant? If `num_deltas` is 0, `blkcache_deltas` should remain `NULL`. And we can still call `__wt_free` on a `NULL`. I think it would make more sense ot have an assert that `blkcache_deltas` is NULL whenever `num_deltas` is 0. What do you think @ershov and @korteland ?